src/kOS.Safe/Exceptions/KOSArgumentMismatchException.cs

```diff
@@ -41,15 +49,16 @@ public override string HelpURL
         public KOSArgumentMismatchException(string message = "") :
             base( BuildTerseMessage() + " " + message )
         {
-            expectedNum = 0;
-            actualNum = 0;
         }
         
-        private static string BuildTerseMessage(int expected, int actual)
+        private static string BuildTerseMessage(IList<int> expected, int actual)
         {
-            return String.Format("Incorrect number of arguments.  Expected {0} argument{1}, but found {2}",
-                                 (expected==0?"no":expected.ToString()), (expected==1?"":"s"), (actual==0?"none":actual.ToString())
-                                );
+            var expectedDisplay = (expected.Any() ? String.Join(", ", new List<int>(expected).ConvertAll(i => i.ToString()).ToArray()) : "no");
+            var pluralDecorator = (expected.Count() == 1 ? "" : "s");
+            var actualArgs = (actual == 0 ? "none" : actual.ToString());
+
+            return string.Format("Incorrect number of arguments.  Expected {0} argument{1}, but found {2}",
```

Comment:
This error message should probably end with  a dot because `message` can be appended to it's end.